core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -651,60 +651,126 @@ class UserService(protected val ctx: RawlsRequestContext,
       }
     }
 
+  private def getLegacyBillingPolicies(samRole: ProjectRole): Seq[SamResourcePolicyName] =
+    samRole match {
+      case ProjectRoles.Owner => Seq(SamBillingProjectPolicyNames.owner)
+      case ProjectRoles.User =>
+        Seq(SamBillingProjectPolicyNames.workspaceCreator, SamBillingProjectPolicyNames.canComputeUser)
+    }
+  private def getV2BillingPolicy(samRole: ProjectRole): SamResourcePolicyName =
+    samRole match {
+      case ProjectRoles.Owner => SamBillingProjectPolicyNames.owner
+      case ProjectRoles.User  => SamBillingProjectPolicyNames.workspaceCreator
+    }
+
   def addUserToBillingProject(projectName: RawlsBillingProjectName,
                               projectAccessUpdate: ProjectAccessUpdate
   ): Future[Unit] =
     requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
-      val policies = projectAccessUpdate.role match {
-        case ProjectRoles.Owner => Seq(SamBillingProjectPolicyNames.owner)
-        case ProjectRoles.User =>
-          Seq(SamBillingProjectPolicyNames.workspaceCreator, SamBillingProjectPolicyNames.canComputeUser)
+      val policies = getLegacyBillingPolicies(projectAccessUpdate.role)
+      addUserToBillingProjectInner(projectName, projectAccessUpdate, policies)
+    }
+
+  private def addUserToBillingProjectInner(projectName: RawlsBillingProjectName,
+                                           projectAccessUpdate: ProjectAccessUpdate,
+                                           policies: Seq[SamResourcePolicyName]
+  ): Future[Unit] =
+    for {
+      _ <- Future.traverse(policies) { policy =>
+        samDAO
+          .addUserToPolicy(SamResourceTypeNames.billingProject,
+                           projectName.value,
+                           policy,
+                           projectAccessUpdate.email,
+                           ctx.userInfo
+          )
+          .recoverWith { case regrets: Throwable =>
+            if (policy == SamBillingProjectPolicyNames.canComputeUser) {
+              logger.info(
+                s"error adding user to canComputeUser policy for $projectName likely because it is a v2 billing project which does not have a canComputeUser policy. regrets: ${regrets.getMessage}"
+              )
+              Future.successful(())
+            } else {
+              Future.failed(regrets)
+            }
+          }
       }
+    } yield {}
 
+  def addUserToBillingProjectV2(projectName: RawlsBillingProjectName,
+                                projectAccessUpdate: ProjectAccessUpdate
+  ): Future[Unit] =
+    requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
```

Comment:
I'm not seeing the reason for this method to exist; it looks like it's the same behavior as `addUserToBillingProject` with the difference being that it's checking for a related billing profile ; if present it's adding the user to the billing profile and using the newer policies for the call to `adduserToBillingProjectInner`. 

Why wouldn't we want this behavior on both the v2 and v1 apis as we have it today? I admit it's odd that we have an existing v2 API that's calling the same codepath as the v1 api.

Reply:
I thought our general philosophy was that only the v2 APIs supported BPM. For example, we defined a `listBillingProjectsV2` method that gets things from BPM, and left the existing `listBillingProjects` method alone.

I'm also trying to line up with the operationId defined in Swagger by naming the method `addUserToBillingProjectV2`.

Reply:
We have decided to keep calls to BPM in the v2 versions only. I still need to see if I can mark the v1 methods as deprecated in the Swagger.

Reply:
Have marked endpoints deprecated in Swagger.